core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -267,12 +279,17 @@ private[spark] class EventLoggingListener(
 
   override def onExecutorMetricsUpdate(event: SparkListenerExecutorMetricsUpdate): Unit = {
     if (shouldLogStageExecutorMetrics) {
-      // For the active stages, record any new peak values for the memory metrics for the executor
-      event.executorUpdates.foreach { executorUpdates =>
-        liveStageExecutorMetrics.values.foreach { peakExecutorMetrics =>
-          val peakMetrics = peakExecutorMetrics.getOrElseUpdate(
-            event.execId, new ExecutorMetrics())
-          peakMetrics.compareAndUpdatePeakValues(executorUpdates)
+      event.executorUpdates.foreach { case (k1, peakUpdates) =>
+        liveStageExecutorMetrics.foreach { case (k2, peakExecutorMetrics) =>
+          // If the update came from the driver, the key k1 will be the dummy key (-1, -1),
+          // so record those peaks for all active stages.
+          // Otherwise, record the peaks for the matching stage.
+          val k0 = (-1, -1)
```

Comment:
How about more descriptive names, like driverStageKey instead of k0, maybe stageKey1 instead of k1, and stageKey2 instead of k2?